benches/bench.rs

```diff
@@ -0,0 +1,76 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+
+use rand::thread_rng;
+use redjubjub::*;
+use std::convert::TryFrom;
+
+fn sigs_with_distinct_pubkeys<T: SigType>(
+) -> impl Iterator<Item = (VerificationKeyBytes<T>, Signature<T>)> {
+    std::iter::repeat_with(|| {
+        let sk = SigningKey::<T>::new(thread_rng());
+        let vk_bytes = VerificationKey::from(&sk).into();
+        let sig = sk.sign(thread_rng(), b"");
+        (vk_bytes, sig)
+    })
+}
+
+fn sigs_with_same_pubkey<T: SigType>(
+) -> impl Iterator<Item = (VerificationKeyBytes<T>, Signature<T>)> {
+    let sk = SigningKey::<T>::new(thread_rng());
+    let vk_bytes = VerificationKey::from(&sk).into();
+    std::iter::repeat_with(move || {
+        let sig = sk.sign(thread_rng(), b"");
+        (vk_bytes, sig)
+    })
+}
```

Comment:
In the ed25519-zebra code this is adapted from, there's an optimization for verifying many signatures with the same public key, but since that usecase doesn't make sense for Redjubjub and the optimization shouldn't be present in this implementation, I think it doesn't make sense to benchmark "distinct pubkeys" and "same pubkeys" separately, so the same-pubkey benchmark should be removed.

We might want to also change "pubkey" to "verification key", but if the distinction above is removed there's no need to refer to pubkeys or verification keys at all ("unbatched verification" vs "batched verification").